app/modules/topics/components/NewTopicCard.js

```diff
@@ -6,25 +6,48 @@ import { Field, reduxForm } from 'redux-form';
 import type { FormProps } from 'redux-form';
 import { translate } from 'react-i18next';
 import type { CustomTranslatorProps } from 'types/translator';
+import type { State } from 'types/state';
+import type { Identifier } from 'types/model';
 import { Link, withRouter } from 'react-router-dom';
 import { Form, Button } from 'semantic-ui-react';
-import { CURRENT_USER } from 'modules/users/constants';
 import type { RouterHistory } from 'react-router-dom';
+import authentication from 'modules/authentication';
 import { add } from '../actions';
 
+const { getAccount } = authentication.selectors;
 
 type DispatchProps = {
-  onAddButtonClick: (string, string, RouterHistory) => void,
+  onAddButtonClick: (Identifier, string, string, RouterHistory) => void,
+};
+
+type StateProps = {
+  userId: Identifier,
 };
 
 type Props = CustomTranslatorProps & DispatchProps & FormProps & RouterHistory;
 
+const mapStateToProps = (state: State): StateProps => {
+  const account = getAccount(state);
+
+  // TODO: does this need null checks or is it impossible to access when not logged in?
```

Comment:
`AuthState` has a property `authenticated` (boolean) which should be used to determine whether or not the user is logged in. It should be used to determine whether or not to redirect.

Reply:
But that's not the component's responsibility. So the redirection should happen at the Page level. Null checks will be removed when unauthorised users can't access this anymore.

Reply:
Agreed, so neither are the null checks. Authenticated components should assume there is a current user.